fairseq/data/indexed_dataset.py

```diff
@@ -15,9 +15,16 @@
 from . import FairseqDataset
 
 
-def make_builder(out_file, impl):
+def __best_fitting_dtype(vocab_size=None):
+    if vocab_size is not None and vocab_size < 65500:
+        return np.uint16
+    else:
+        return np.int32
```

Comment:
This makes sense. Is this the main cause of the memory/speed improvement?

Reply:
Yes, this is actual boost. I now have a model that is 1/4th of the original and this means a huge saving in memory, disk and also load time!